packages/react/src/side-nav/SideNavLink.tsx

```diff
@@ -70,7 +75,25 @@ export function SideNavLink({
 				}}
 			>
 				<SideNavLinkIndicator depth={depth} />
-				<span>{label}</span>
+
+				<span css={{ flex: 1 }}>{label}</span>
+
+				{Boolean(children) &&
```

Comment:
It's a legacy API, but should we be using `isValidElement()`? https://react.dev/reference/react/isValidElement